components/datetime/src/format.rs

```diff
@@ -230,6 +231,18 @@ where
             ) as isize,
             field.length,
         )?,
+        FieldSymbol::Second(Second::FractionalSecond) => format_number(
+            w,
+            usize::from(
+                date_time
+                    .date_time()
+                    .fraction()
+                    .map(|f| f.milliseconds())
+                    .ok_or(Error::MissingInputField)?,
+            ) as isize,
+            field.length,
```

Comment:
issue: The syntax is like `ss.SSS`.  The number of `S`s means how many fractional digits to display.  For example:

- `S` = tenths of a second
- `SS` = hundredths of a second
- `SSS` = thousandths of a second (milliseconds)
- `SSSS` = ten-thousandths of a second
- ...

So, I think it's not correct to take `f.milliseconds()` here.  We need to take a number of digits equal to the field length.